storage/interface.go

```diff
@@ -223,6 +223,17 @@ type SelectHints struct {
 	// When disabled, the result may contain samples outside the queried time range but Select() performances
 	// may be improved.
 	DisableTrimming bool
+
+	// Projection hints. They are currently unused in the Prometheus promql engine but can be used by different
+	// implementations of the Queryable interface and engines.
+
+	// ProjectionLabels are the minimum amount of labels required to be fetched for this Select call
```

Comment:
Can you give more hints how PromQL engines might know that minimum labels?

Reply:
Added an example to docs